geode-assembly/src/acceptanceTest/resources/ManagementClientCreateRegion.java

```diff
@@ -32,14 +33,18 @@ public static void main(String[] args) throws Exception {
     if (useSsl) {
       // The default SSLContext will pull in all necessary javax.net.ssl properties
       cms =
-          new ClusterManagementServiceBuilder().setConnectionConfig(
-              new BaseConnectionConfig("localhost", httpPort)
-                  .setSslContext(SSLContext.getDefault()))
+          new ClusterManagementServiceBuilder()
+              .setTransport(new RestTemplateClusterManagementServiceTransport(
```

Comment:
Is there anyway to make building cms easier? I don't need to know any RestTemplateClusterManagementServiceTransport, I don't need to know any BaseConnectionConfig, I just want to plug in the host and port.....

Reply:
host and port can be set via system properties.
In this change all that I have done is made the transport explicit. I don't think there is anything wrong with exposing that there is a Transport and what kind of transport is being used.

What is incorrect is having multiple options on a Builder that if you incorrectly pick will cause failure, rather than an intuitive Builder where differences from "normal"  are handled as optional and there is a happy case.

Reply:
what I mean is: is there a way to build a cms without having to know I need to construct a BaseConnectionConfig and a a RestTemplate....Transport.

Reply:
@jinmeiliao I've made a few changes. Added `GeodeClusterManagementServiceBuilder` which is specifically geared towards the Geode Cache CMS configuration. The current `ClusterManagementServiceBuilder` is specifically targeted to a non-Cache CMS builder.

I've also added `GeodeConnectionConfig` and cleaned up the `BaseConnectionConfig` / `ConnectionConfig` confusion.

I hope that my latest refactor will not force the user to know about `RestTemplate..... Transport`. You can still create your own `Transport` and add it using `setTransport`